sensorlib/src/CarlaCDASimAPI.py

```diff
@@ -108,30 +106,45 @@ def create_simulated_semantic_lidar_sensor(self, simulated_sensor_config, carla_
         # Retrieve the CARLA sensor
         blueprint_library = self.__carla_world.get_blueprint_library()
         sensor_bp = self.__generate_lidar_bp(blueprint_library, carla_sensor_config)
-        parent_actor = CarlaUtils.get_actor(self.__carla_world, parent_actor_id)
-        carla_sensor = self.__carla_world.spawn_actor(sensor_bp, sensor_transform, attach_to=parent_actor)
-        self.__carla_world.wait_for_tick()
+        parent = None
+        if parent_id is not None:
+            parent = self.__carla_world.get_actor(parent_id)
+        carla_sensor = self.__carla_world.spawn_actor(sensor_bp, sensor_transform, parent)
+
+        # Fix for CARLA not updating position immediately
+        sleep(0.2)
+
+        print(f"CarlaCDASimAPI: Creating sensor in CARLA at sensor_position: {carla_sensor.get_location()}")
 
         # Build internal objects
         sensor = CarlaSensorBuilder.build_sensor(carla_sensor)
-        data_collector = SensorDataCollector(self.__carla_world, carla_sensor)
+        data_collector = SensorDataCollector(self.__carla_world, carla_sensor, custom_callback)
         noise_model = NoiseModelFactory.get_noise_model(noise_model_config["noise_model_name"], noise_model_config)
 
         # Construct the SimulatedSensor
         simulated_sensor = SemanticLidarSensor(infrastructure_id, sensor_id, simulated_sensor_config,
                                                carla_sensor_config,
                                                self.__carla_world, sensor,
-                                               data_collector, noise_model)
+                                               data_collector, noise_model,
+                                               parent_id)
 
         # Register the sensor
         self.__infrastructure_sensors[(infrastructure_id, sensor_id)] = simulated_sensor
 
+        sleep(0.5)
```

Comment:
Why are we sleeping here?

Reply:
it was not obvious, so removed and works without it